### PR TITLE
Rational - macro support

### PR DESCRIPTION
**Pull request by sibartlett**

### CODE REVIEW

File: sdetools/modules/sync_rational/rational_plugin.py [hunk 1]
```diff
@@ -364,21 +357,20 @@ def alm_connect_project(self):
 
         self._get_resource_shapes()
 
-        self.priorities = self._rational_get_priorities()
-
     def alm_validate_configurations(self):
 
+        return
```
Comment (sibartlett) [inline sdetools/modules/sync_rational/rational_plugin.py]: Need to remove `return` and fix this method.

**Comment by shawngustaw** [comment]
NTS: remove custom lookup fields.

File: sdetools/modules/sync_rational/rational_plugin.py [hunk 2]
```diff
@@ -139,14 +145,16 @@ def get_timestamp(self):
...
+class RationalConnector(MacroAlmConnector):
 
     alm_name = 'Rational Team Concert'
     cm_service_provider = None
     resource_url = None
     priorities = None
     ALM_DONE_STATUSES = 'rational_done_statuses'
     default_priority_map = RATIONAL_DEFAULT_PRIORITY_MAP
+    default_issue_template = RATIONAL_DEFAULT_ISSUE_TEMPLATE
+    feature_custom_lookup = True
```
Comment (geoffwhittington) [inline sdetools/modules/sync_rational/rational_plugin.py]: need to make this `False` if we no longer support this. Better yet, just remove this line.

File: sdetools/modules/sync_rational/rational_plugin.py [hunk 3]
```diff
@@ -389,9 +379,11 @@ def alm_get_task(self, task):
         if not task_id:
             return None
 
+        conditions = ['dcterms:title="%s*"' % task['alm_fixed_title']]
+
+        oslc_where = urlencode_str(' and ').join(conditions)
         try:
-            work_items = self._call_api('%s/workitems?oslc.where=dcterms:title="%s*"' % (
-                                        self.query_url, urlencode_str(task['alm_fixed_title'])))
+            work_items = self._call_api('%s/workitems?oslc.where=%s' % (self.query_url, oslc_where))
```
Comment (shawngustaw) [inline sdetools/modules/sync_rational/rational_plugin.py]: I think this change can be undone now as well. Just noticed it

Comment (sibartlett) [inline sdetools/modules/sync_rational/rational_plugin.py]: Yep, I agree we can undo this.

**Pull request merged**